### ISSUE TITLE
Doubt in the example of classification using DynamicEdgeConv

### ISSUE THREAD

**Issue by dhorka** [rusty1s/pytorch_geometric#553]
I was reviewing the example of dynamic edge conv. I'm not sure if the aim of this example is to reproduce the results of the paper. However, I found that I think it is a discrepancy between the implemented architecture and the archicture showed in the paper in the case of classification. In the example it seems like the mlp used have more than one layer and in the paper as it showed in the following image only one layer is used in the mlp.

`self.conv1 = DynamicEdgeConv(MLP([2 * 3, 64, 64, 64]), k, aggr)`

![image](https://user-images.githubusercontent.com/39189267/61711646-cfff8b80-ad54-11e9-8660-9f0233bcfa10.png)

**Comment by rusty1s** [comment]
Good catch! It seems that the paper got updated and our implementation refers to [v1](https://arxiv.org/pdf/1801.07829v1.pdf). Are you interested in updating the example architecture?

**Comment by dhorka** [comment]
I think we need to wait. Because I was checking their implementation and there is a miss-match between the paper and the implementation. For instance, in the paper in the mlp they said that their are using a ReLU and in the implementation their are using a leaky_relu.

**Comment by rusty1s** [comment]
Sounds good to me :)

**Comment by cy69855522** [comment]
Hi, @dhorka.  I added a dgcnnV2 segmentation example, maybe it's helpful.

**Comment by liangzhenm** [comment]
@cy69855522 哥们为什么我试运行你的代码，显示：    
pos, batch = data.pos, data.batch
AttributeError: 'list' object has no attribute 'pos'

**Comment by cy69855522** [comment]
@dhorka 建议把完整代码拷贝后测试，我试过没问题。这份代码是用在多GPU上的（单GPU也行），如果你只拷贝一部分的话可能会报错。如果你只是想整合进自己的代码，只需要修改数据增强和Net类定义的部分。

**Comment by liangzhenm** [comment]
@cy69855522 非常感谢！！我直接拷贝了你的全部代码，能够运行，不出现上面那个错误，但是又出现了了：c = torch.cat([data.category for data in data_list])
AttributeError: 'Data' object has no attribute 'category' 的错误。
由于PYG官网上给的shapenet网址的数据集现在好像不开放了，只能下载到训练集和验证集，下载不到测试集的label，所以我就把验证集当做了测试集，想试验一下，然而却爆出这个错误，请问这是因为我数据的问题吗？

**Comment by liangzhenm** [comment]
@cy69855522 你方便的话，能传我一份shapenet数据集吗？邮箱、百度盘都可以，我邮箱liangzhm@163.com  非常感谢！！

**Comment by cy69855522** [comment]
> @cy69855522 你方便的话，能传我一份shapenet数据集吗？邮箱、百度盘都可以，我邮箱[liangzhm@163.com](mailto:liangzhm@163.com) 非常感谢！！

有开放呀，我刚刚重新下载了一次都可以。你可以把ShapeNet这个文件夹删除掉，直接跑代码，它会自动整理，重新下载的。

**Comment by liangzhenm** [comment]
> > @cy69855522 你方便的话，能传我一份shapenet数据集吗？邮箱、百度盘都可以，我邮箱[liangzhm@163.com](mailto:liangzhm@163.com) 非常感谢！！
> 
> 有开放呀，我刚刚重新下载了一次都可以。你可以把ShapeNet这个文件夹删除掉，直接跑代码，它会自动整理，重新下载的。

我知道会自动下载，但是每次都是下载一点，就显示网络不可达，下载失败，是我自己网的问题？？应该不是啊，我实验室网络很好啊，还是墙的问题？？

**Comment by cy69855522** [comment]
@dhorka 不知道诶，我在国外。百度云7天有效。链接: https://pan.baidu.com/s/1BIji0KXnQotNeU8HGdpXOQ 提取码: pur6 复制这段内容后打开百度网盘手机App，操作更方便哦

**Comment by cy69855522** [comment]
> > > @cy69855522 你方便的话，能传我一份shapenet数据集吗？邮箱、百度盘都可以，我邮箱[liangzhm@163.com](mailto:liangzhm@163.com) 非常感谢！！
> > 
> > 
> > 有开放呀，我刚刚重新下载了一次都可以。你可以把ShapeNet这个文件夹删除掉，直接跑代码，它会自动整理，重新下载的。
> 
> 我知道会自动下载，但是每次都是下载一点，就显示网络不可达，下载失败，是我自己网的问题？？应该不是啊，我实验室网络很好啊，还是墙的问题？？

应该是墙的问题吧，我在国外。百度云7天。
链接:https://yun.baidu.com/s/1BIji0KXnQotNeU8HGdpXOQ 提取码:pur6 复制这段内容后打开百度网盘手机App，操作更方便哦

**Comment by liangzhenm** [comment]
> > > > @cy69855522 你方便的话，能传我一份shapenet数据集吗？邮箱、百度盘都可以，我邮箱[liangzhm@163.com](mailto:liangzhm@163.com) 非常感谢！！
> > > 
> > > 
> > > 有开放呀，我刚刚重新下载了一次都可以。你可以把ShapeNet这个文件夹删除掉，直接跑代码，它会自动整理，重新下载的。
> > 
> > 
> > 我知道会自动下载，但是每次都是下载一点，就显示网络不可达，下载失败，是我自己网的问题？？应该不是啊，我实验室网络很好啊，还是墙的问题？？
> 
> 应该是墙的问题吧，我在国外。百度云7天。
> 链接:https://yun.baidu.com/s/1BIji0KXnQotNeU8HGdpXOQ 提取码:pur6 复制这段内容后打开百度网盘手机App，操作更方便哦

实在是太感谢了！
但是我试了一下，当train完一个epoch还是出现上面那个
    c = torch.cat([data.category for data in data_list])
AttributeError: 'Data' object has no attribute 'category'
的问题。。。我每次都遇到这个问题不知道为什么

**Comment by cy69855522** [comment]
@liangzhenm 可能是pyg包没更新，pip的不是最新版，可以去github拷贝

**Comment by liangzhenm** [comment]
> @liangzhenm 可能是pyg包没更新，pip的不是最新版，可以去github拷贝

我重新源码安装了PYG，果然可以了！谢谢谢谢，这个问题困扰我很久了，今天终于弄懂了，非常感谢，真的是麻烦你了！
